util/scripts/launch.ps1

```diff
@@ -29,9 +29,5 @@ $base="$PSScriptRoot\..\..\"
 $gradlew="${base}/gradlew.bat"
 
 # invoke script
-if ($args.Length -eq 0) {
-    & "${gradlew}" -p "${base}" "cli:${tool}:run"
-} else {
-    $argsString = $args -join " "
-    & "${gradlew}" -p "${base}" "cli:${tool}:run" --args="${argsString}"
-}
\ No newline at end of file
+& "${gradlew}" assemble
```

Comment:
I wonder how optimized the "run" task it. The namespace suggests that it would only assemble the required tool, not all all three. Yet again, I doubt there would be a noticeable difference because they all rely on the same main code base, which will only be compiled once... It's arguably simpler and it separates building from running, so I think it's an improvement. @cmnrd, WDYT?

Reply:
If we want to build only one target we could do `./gradlew :cli:${tool}:assemble`, which is as optimised as `:cli:lfc:run`.
<details>
  <summary>Click to reveil dry run results</summary>
  
```bash
gitpod /workspace/playground-lingua-franca/lingua-franca (master) $ ./gradlew --console=verbose --dry-run -p . "cli:lfc:run" --args="--help"
Picked up JAVA_TOOL_OPTIONS:  -Xmx12884m
> Task :buildSrc:extractPluginRequests UP-TO-DATE
> Task :buildSrc:generatePluginAdapters UP-TO-DATE
> Task :buildSrc:compileJava UP-TO-DATE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:compileGroovyPlugins UP-TO-DATE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources UP-TO-DATE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
:core:generateGrammarSource SKIPPED
:core:generateIntegrationTestGrammarSource SKIPPED
:core:generateTestFixturesGrammarSource SKIPPED
:core:generateTestGrammarSource SKIPPED
:core:generateXtextLanguage SKIPPED
:core:compileKotlin SKIPPED
:core:compileJava SKIPPED
:core:processResources SKIPPED
:core:classes SKIPPED
:core:inspectClassesForKotlinIC SKIPPED
:core:jar SKIPPED
:cli:base:compileKotlin SKIPPED
:cli:base:compileJava SKIPPED
:cli:base:processResources SKIPPED
:cli:base:classes SKIPPED
:cli:base:inspectClassesForKotlinIC SKIPPED
:cli:base:jar SKIPPED
:cli:lfc:compileKotlin SKIPPED
:cli:lfc:compileJava SKIPPED
:cli:lfc:processResources SKIPPED
:cli:lfc:classes SKIPPED
:cli:lfc:run SKIPPED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/8.1.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1s
7 actionable tasks: 7 up-to-date
gitpod /workspace/playground-lingua-franca/lingua-franca (master) $ ./gradlew --console=verbose --dry-run assemble
Picked up JAVA_TOOL_OPTIONS:  -Xmx12884m
> Task :buildSrc:extractPluginRequests UP-TO-DATE
> Task :buildSrc:generatePluginAdapters UP-TO-DATE
> Task :buildSrc:compileJava UP-TO-DATE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:compileGroovyPlugins UP-TO-DATE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources UP-TO-DATE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
:core:generateGrammarSource SKIPPED
:core:generateIntegrationTestGrammarSource SKIPPED
:core:generateTestFixturesGrammarSource SKIPPED
:core:generateTestGrammarSource SKIPPED
:core:generateXtextLanguage SKIPPED
:core:compileKotlin SKIPPED
:core:compileJava SKIPPED
:core:processResources SKIPPED
:core:classes SKIPPED
:core:inspectClassesForKotlinIC SKIPPED
:core:jar SKIPPED
:cli:base:compileKotlin SKIPPED
:cli:base:compileJava SKIPPED
:cli:base:processResources SKIPPED
:cli:base:classes SKIPPED
:cli:base:inspectClassesForKotlinIC SKIPPED
:cli:base:jar SKIPPED
:cli:lfc:compileKotlin SKIPPED
:cli:lfc:compileJava SKIPPED
:cli:lfc:processResources SKIPPED
:cli:lfc:classes SKIPPED
:cli:lfc:inspectClassesForKotlinIC SKIPPED
:cli:lfc:jar SKIPPED
:cli:lfc:startScripts SKIPPED
:cli:lfc:installDist SKIPPED
:cli:lfd:compileJava SKIPPED
:cli:lfd:processResources SKIPPED
:cli:lfd:classes SKIPPED
:cli:lfd:jar SKIPPED
:cli:lfd:startScripts SKIPPED
:cli:lfd:installDist SKIPPED
:cli:lff:compileJava SKIPPED
:cli:lff:processResources SKIPPED
:cli:lff:classes SKIPPED
:cli:lff:jar SKIPPED
:cli:lff:startScripts SKIPPED
:cli:lff:installDist SKIPPED
:clitoolsDistTar SKIPPED
:clitoolsDistZip SKIPPED
:distTar SKIPPED
:distZip SKIPPED
:installClitoolsDist SKIPPED
:installDist SKIPPED
:assemble SKIPPED
:core:assemble SKIPPED
:lsp:compileJava SKIPPED
:lsp:processResources SKIPPED
:lsp:classes SKIPPED
:lsp:jar SKIPPED
:lsp:startScripts SKIPPED
:lsp:distTar SKIPPED
:lsp:distZip SKIPPED
:lsp:shadowJar SKIPPED
:lsp:startShadowScripts SKIPPED
:lsp:shadowDistTar SKIPPED
:lsp:shadowDistZip SKIPPED
:lsp:assemble SKIPPED
:cli:base:assemble SKIPPED
:cli:lfc:distTar SKIPPED
:cli:lfc:distZip SKIPPED
:cli:lfc:assemble SKIPPED
:cli:lfd:distTar SKIPPED
:cli:lfd:distZip SKIPPED
:cli:lfd:assemble SKIPPED
:cli:lff:distTar SKIPPED
:cli:lff:distZip SKIPPED
:cli:lff:assemble SKIPPED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/8.1.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1s
7 actionable tasks: 7 up-to-date
gitpod /workspace/playground-lingua-franca/lingua-franca (master) $ ./gradlew --console=verbose --dry-run :cli:lfc:assemble
Picked up JAVA_TOOL_OPTIONS:  -Xmx12884m
> Task :buildSrc:extractPluginRequests UP-TO-DATE
> Task :buildSrc:generatePluginAdapters UP-TO-DATE
> Task :buildSrc:compileJava UP-TO-DATE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:compileGroovyPlugins UP-TO-DATE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources UP-TO-DATE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
:core:generateGrammarSource SKIPPED
:core:generateIntegrationTestGrammarSource SKIPPED
:core:generateTestFixturesGrammarSource SKIPPED
:core:generateTestGrammarSource SKIPPED
:core:generateXtextLanguage SKIPPED
:core:compileKotlin SKIPPED
:core:compileJava SKIPPED
:core:processResources SKIPPED
:core:classes SKIPPED
:core:inspectClassesForKotlinIC SKIPPED
:core:jar SKIPPED
:cli:base:compileKotlin SKIPPED
:cli:base:compileJava SKIPPED
:cli:base:processResources SKIPPED
:cli:base:classes SKIPPED
:cli:base:inspectClassesForKotlinIC SKIPPED
:cli:base:jar SKIPPED
:cli:lfc:compileKotlin SKIPPED
:cli:lfc:compileJava SKIPPED
:cli:lfc:processResources SKIPPED
:cli:lfc:classes SKIPPED
:cli:lfc:inspectClassesForKotlinIC SKIPPED
:cli:lfc:jar SKIPPED
:cli:lfc:startScripts SKIPPED
:cli:lfc:distTar SKIPPED
:cli:lfc:distZip SKIPPED
:cli:lfc:assemble SKIPPED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/8.1.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 860ms
7 actionable tasks: 7 up-to-date
```
</details>

Reply:
Also, we might want to limit the output of Gradle as much as possible...

Reply:
Use the `--quiet` flag then...... lemme reflect all of these